week-2/Homework/mandatory/4-sorting-algorithm.js

```diff
@@ -13,45 +13,85 @@ sort method 😎)
 You don't have to worry about making this algorithm work fast! The idea is to get you to
 "think" like a computer and practice your knowledge of basic JavaScript.
 */
+function filterArrays(arr) {
+  let filterArray = [];
 
-function sortAges(arr) {
+  for (let i = 0; i < arr.length; i++) {
+    if (typeof arr[i] === "number") {
+      filterArray.push(arr[i]);
+    }
+  }
+  return filterArray;
+}
 
+function sortAges(arr) {
+  let filterArray = filterArrays(arr);
+  let sortedArray = filterArray;
+  let swamp;
+  do {
+    swamp = false;
+    for (let i = 0; i < sortedArray.length - 1; i++) {
+      if (sortedArray[i] > sortedArray[i + 1]) {
+        //swamp
+        let temp = sortedArray[i];
+        sortedArray[i] = sortedArray[i + 1];
+        sortedArray[i + 1] = temp;
+        swamp = true;
+      }
+    }
+  } while (swamp);
+  return sortedArray;

```

Comment:
Ok, this works, nice! 

Some questions you can think about: Do we need the `swamp` variable here? Is there a way we can do this in 2 `for` loops if we do not worry about using unnecessary computation?